pkg/util/installer/kubeadmin.go

```diff
@@ -0,0 +1,31 @@
+package installer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	clientcmdv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+)
+
+// See github.com/openshift/installer/pkg/asset/password
+type KubeadminPasswordData struct {
+	Password string
+}
+
+type AdminKubeConfigSignerCertKey struct {
+	SelfSignedCertKey
+}
+
+// AdminKubeConfigClientCertKey is the asset that generates the key/cert pair for admin client to apiserver.
+type AdminKubeConfigClientCertKey struct {
+	SignedCertKey
```

Comment:
what are those for? if this is just to get the nested fields in json I think we should be explicit with json tags and maybe a comment. Embedding can be a little confusing when it's not used to extend objects/interfaces

Reply:
part of this is to maintain parity with upstream, even if it's not super nice